core/win32/callback.c

```diff
@@ -4976,14 +4976,30 @@ check_internal_exception(dcontext_t *dcontext, CONTEXT *cxt,
         if (is_in_dynamo_dll((app_pc)pExcptRec->ExceptionAddress))
             is_DR_exception = true;
         else {
+            /*
+             * i#2144 : We exclude single step cases in a jump.
+             */
+            if (pExcptRec->ExceptionCode == EXCEPTION_SINGLE_STEP &&
+                !in_fcache(pExcptRec->ExceptionAddress)) {
```

Comment:
The comment below is now in the wrong place, about calling in_fcache last.

Logically this would be better out in the caller and not here in check_internal_exception which is just supposed to handle a crash in DR itself.

Probably what is needed is a query in_exit_stub().